packages/next/src/client/components/navigation.ts

```diff
@@ -140,6 +168,10 @@ export function useRouter(): AppRouterInstance {
  */
 // Client components API
 export function useParams<T extends Params = Params>(): T {
+  // If there are any unknown route parameters, then we should track this as
+  // a dynamic access.
```

Comment:
this comment doesn't seem relevant to the callsite, just the implementation, as `trackParamsAccessed` is called unconditionally.